desc/coils.py

```diff
@@ -1609,6 +1631,13 @@ def _compute_A_or_B(
             ValueError,
             f'Expected "A" or "B" for compute_A_or_B, instead got {compute_A_or_B}',
         )
+        # NFP symmetry applies to coilset as a whole, not individual coils, so the grid
+        # should have NFP=1.
+        errorif(
+            getattr(source_grid, "NFP", 1) != 1,
+            ValueError,
+            "source_grid for CoilSet must have NFP=1",
+        )
```

Comment:
If you scroll down, the nfp loop for the biot-savart uses ``self.NFP`` not source_grid.NFP. it may be useful to be defensive and not pass in the source_grid as input to ``op``.

Reply:
Not sure what you mean? That loop should use the coilset NFP

Reply:
The arguments given to the `op` function are ``coords`` which take ``NFP`` as self.NFP``, and ``source_grid``, which may have a different ``NFP`` than the one associated to ``coords``. Can this lead to problems in the future? If so, I don't think ``source_grid`` is used for the ``op`` function anyway, so maybe it's better to not pass it in.